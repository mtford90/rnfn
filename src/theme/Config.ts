import FontWeightName from "./FontWeightName";

type FontWeightMap = Partial<Record<FontWeightName, { fontFamily: string }>>;

export default interface Config {
  theme: {
    colors: Record<string | number, string | Record<string | number, string>>;
    spacing: Record<string, number>;
    fontSize: Record<string, [number, { lineHeight: number }]>;
    fontFamily: Record<
      string,
      Partial<Record<"normal" | "italic", FontWeightMap>>
    >;
  };
}
