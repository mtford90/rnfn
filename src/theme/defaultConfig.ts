import colors from "./colors";
import Config from "./Config";

const defaultConfig: Config = {
  theme: {
    colors: {
      transparent: "transparent",
      black: colors.black,
      white: colors.white,
      gray: colors.coolGray,
      red: colors.red,
      yellow: colors.amber,
      green: colors.green,
      emerald: colors.emerald,
      lime: colors.lime,
      rose: colors.rose,
      blue: colors.blue,
      lightBlue: colors.lightBlue,
      amber: colors.amber,
      orange: colors.orange,
      cyan: colors.cyan,
      teal: colors.teal,
      warmGray: colors.warmGray,
      fuchsia: colors.fuchsia,
      violet: colors.violet,
      indigo: colors.indigo,
      purple: colors.violet,
      pink: colors.pink,
      trueGray: colors.trueGray,
      coolGray: colors.coolGray,
      blueGray: colors.blueGray,
    },
    spacing: {
      dp: 1,
      "0": 0,
      "0.5": 2,
      "1": 4,
      "1.5": 6,
      "2": 8,
      "2.5": 10,
      "3": 12,
      "3.5": 14,
      "4": 16,
      "5": 20,
      "6": 24,
      "7": 28,
      "8": 32,
      "9": 36,
      "10": 40,
      "11": 44,
      "12": 48,
      "14": 52,
      "16": 64,
      "20": 80,
      "24": 96,
      "28": 112,
      "32": 128,
      "36": 144,
      "40": 160,
      "44": 176,
      "48": 192,
      "52": 208,
      "56": 224,
      "60": 240,
      "64": 256,
      "72": 288,
      "80": 320,
      "96": 374,
    },
    fontSize: {
      xs: [12, { lineHeight: 16 }],
      sm: [14, { lineHeight: 20 }],
      base: [16, { lineHeight: 24 }],
      lg: [18, { lineHeight: 28 }],
      xl: [20, { lineHeight: 28 }],
      "2xl": [24, { lineHeight: 32 }],
      "3xl": [30, { lineHeight: 36 }],
      "4xl": [36, { lineHeight: 40 }],
    },
    fontFamily: {},
  },
};

export default defaultConfig;
