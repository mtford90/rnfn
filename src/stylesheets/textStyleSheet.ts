import { StyleSheet, TextStyle } from "react-native";
import defaultConfig from "../theme/defaultConfig";
import { getColorPropValues } from "../props/color";
import { NamedStyles } from "./types";
import {
  getColorStyle,
  getFontFamilyStyle,
  getFontSizeStyle,
} from "../dynamicStyles";

export function getColorNamedStyleKey(colorName: string | number) {
  return `color-${colorName}`;
}

export function getFontSizeNamedStyleKey(fontSizeName: string | number) {
  return `text-${fontSizeName}`;
}

export function getFontFamilyNamedStyleKey(
  style: string | number,
  familyName: string | number,
  weight: string | number
) {
  return `font-family-${style}-${familyName}-${weight}`;
}

export function getTextAlignmentNamedStyleKey(
  alignment: "auto" | "left" | "right" | "center" | "justify" | undefined
) {
  return `align-${alignment}`;
}

export function getTextTransformNamedStyleKey(
  transform: "none" | "capitalize" | "uppercase" | "lowercase" | undefined
) {
  return `transform-${transform}`;
}

export function getTextDecorationLineNamedStyleKey(
  line:
    | "none"
    | "underline"
    | "line-through"
    | "underline line-through"
    | undefined
) {
  return `line-${line}`;
}

export function getNamedColorStyles(config = defaultConfig) {
  const namedStyles: NamedStyles = {};
  const colorPropValues = getColorPropValues(config);

  Object.entries(colorPropValues).forEach(([propValue, colorCode]) => {
    namedStyles[getColorNamedStyleKey(propValue)] = getColorStyle(colorCode);
  });

  return namedStyles;
}

function getNamedFontSizeStyles(config = defaultConfig) {
  const namedStyles: NamedStyles = {};

  Object.entries(config.theme.fontSize).forEach(
    ([propName, [fontSize, props]]) => {
      namedStyles[getFontSizeNamedStyleKey(propName)] = getFontSizeStyle(
        fontSize,
        props
      );
    }
  );

  return namedStyles;
}

function getNamedFontFamilyStyles(config = defaultConfig) {
  const namedStyles: NamedStyles = {};

  Object.entries(config.theme.fontFamily).forEach(([familyName, styles]) => {
    Object.entries(styles).forEach(([style, weights = {}]) => {
      Object.entries(weights).forEach(([weight, definition]) => {
        if (definition) {
          namedStyles[
            getFontFamilyNamedStyleKey(style, familyName, weight)
          ] = getFontFamilyStyle({
            family: definition.fontFamily,
          });
        }
      });
    });
  });

  return namedStyles;
}

function getNamedTextAlignmentStyles() {
  const namedStyles: NamedStyles = {};

  const alignments: Array<TextStyle["textAlign"]> = [
    "auto",
    "left",
    "right",
    "center",
    "justify",
  ];

  alignments.forEach((alignment) => {
    namedStyles[getTextAlignmentNamedStyleKey(alignment)] = {
      textAlign: alignment,
    };
  });

  return namedStyles;
}

function getNamedTextTransformStyles() {
  const namedStyles: NamedStyles = {};

  const transforms: Array<TextStyle["textTransform"]> = [
    "none",
    "capitalize",
    "uppercase",
    "lowercase",
  ];

  transforms.forEach((transform) => {
    namedStyles[getTextTransformNamedStyleKey(transform)] = {
      textTransform: transform,
    };
  });

  return namedStyles;
}

function getNamedTextDecorationLineStyles() {
  const namedStyles: NamedStyles = {};

  const decorationLines: Array<TextStyle["textDecorationLine"]> = [
    "none",
    "underline",
    "line-through",
    "underline line-through",
  ];

  decorationLines.forEach((line) => {
    namedStyles[getTextDecorationLineNamedStyleKey(line)] = {
      textDecorationLine: line,
    };
  });

  return namedStyles;
}

export function getTextStyleSheet(config = defaultConfig) {
  const namedStyles: NamedStyles = {
    ...getNamedColorStyles(config),
    ...getNamedFontSizeStyles(config),
    ...getNamedFontFamilyStyles(config),
    ...getNamedTextAlignmentStyles(),
    ...getNamedTextTransformStyles(),
    ...getNamedTextDecorationLineStyles(),
  };

  return StyleSheet.create(namedStyles);
}
