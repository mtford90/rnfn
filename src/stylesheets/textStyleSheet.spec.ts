import { getTextStyleSheet } from "./textStyleSheet";
import colors from "../theme/colors";
import Config from "../theme/Config";
import defaultConfig from "../theme/defaultConfig";

describe("text stylesheet", () => {
  describe("with default config", () => {
    it("should match the snapshot", async () => {
      const stylesheet = getTextStyleSheet();
      expect(stylesheet).toMatchSnapshot();
    });

    it("should match colours correctly", () => {
      const stylesheet = getTextStyleSheet();
      expect(stylesheet.bgGreen).toEqual({
        backgroundColor: colors.green["500"],
      });
      expect(stylesheet.bgGreen50).toEqual({
        backgroundColor: colors.green["50"],
      });
      expect(stylesheet.colorGreen).toEqual({
        color: colors.green["500"],
      });
      expect(stylesheet.colorGreen50).toEqual({
        color: colors.green["50"],
      });
    });

    it("should add fontsizes", async () => {
      const stylesheet = getTextStyleSheet();

      expect(stylesheet.text2Xl).toEqual({
        fontSize: defaultConfig.theme.fontSize["2xl"][0],
        lineHeight: defaultConfig.theme.fontSize["2xl"][1].lineHeight,
      });
    });
  });

  describe("with a custom config", () => {
    const customConfig: Config = {
      theme: {
        colors: {
          customColor: {
            50: "#fff",
            DEFAULT: "#ccc",
          },
        },
        spacing: {
          lg: 50,
        },
        fontSize: {
          customFontSize: [21, { lineHeight: 19 }],
        },
        fontFamily: {
          customFont: {
            normal: { fontFamily: "customFont" },
            thin: { fontFamily: "customFont-Thin" },
            bold: { fontFamily: "customFont-Bold" },
          },
        },
      },
    };

    it("should match snapshot", async () => {
      const styles = getTextStyleSheet(customConfig);
      expect(styles).toMatchSnapshot();
    });
  });
});
