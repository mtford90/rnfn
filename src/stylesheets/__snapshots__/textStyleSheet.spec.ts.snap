// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text stylesheet with a custom config should match snapshot 1`] = `
Object {
  "alignAuto": Object {
    "textAlign": "auto",
  },
  "alignCenter": Object {
    "textAlign": "center",
  },
  "alignJustify": Object {
    "textAlign": "justify",
  },
  "alignLeft": Object {
    "textAlign": "left",
  },
  "alignRight": Object {
    "textAlign": "right",
  },
  "colorCustomColor": Object {
    "color": "#ccc",
  },
  "colorCustomColor50": Object {
    "color": "#fff",
  },
  "fontFamilyItalicCustomFontBold": Object {
    "fontFamily": "customFont-Italic-Bold",
  },
  "fontFamilyItalicCustomFontNormal": Object {
    "fontFamily": "customFont-Italic",
  },
  "fontFamilyItalicCustomFontThin": Object {
    "fontFamily": "customFont-Italic-Thin",
  },
  "fontFamilyNormalCustomFontBold": Object {
    "fontFamily": "customFont-Bold",
  },
  "fontFamilyNormalCustomFontNormal": Object {
    "fontFamily": "customFont",
  },
  "fontFamilyNormalCustomFontThin": Object {
    "fontFamily": "customFont-Thin",
  },
  "lineLineThrough": Object {
    "textDecorationLine": "line-through",
  },
  "lineNone": Object {
    "textDecorationLine": "none",
  },
  "lineUnderline": Object {
    "textDecorationLine": "underline",
  },
  "lineUnderlineLineThrough": Object {
    "textDecorationLine": "underline line-through",
  },
  "textCustomFontSize": Object {
    "fontSize": 21,
    "letterSpacing": 1.5,
    "lineHeight": 19,
  },
  "transformCapitalize": Object {
    "textTransform": "capitalize",
  },
  "transformLowercase": Object {
    "textTransform": "lowercase",
  },
  "transformNone": Object {
    "textTransform": "none",
  },
  "transformUppercase": Object {
    "textTransform": "uppercase",
  },
}
`;

exports[`text stylesheet with default config should match the snapshot 1`] = `
Object {
  "alignAuto": Object {
    "textAlign": "auto",
  },
  "alignCenter": Object {
    "textAlign": "center",
  },
  "alignJustify": Object {
    "textAlign": "justify",
  },
  "alignLeft": Object {
    "textAlign": "left",
  },
  "alignRight": Object {
    "textAlign": "right",
  },
  "colorAmber": Object {
    "color": "#f59e0b",
  },
  "colorAmber100": Object {
    "color": "#fef3c7",
  },
  "colorAmber200": Object {
    "color": "#fde68a",
  },
  "colorAmber300": Object {
    "color": "#fcd34d",
  },
  "colorAmber400": Object {
    "color": "#fbbf24",
  },
  "colorAmber50": Object {
    "color": "#fffbeb",
  },
  "colorAmber500": Object {
    "color": "#f59e0b",
  },
  "colorAmber600": Object {
    "color": "#d97706",
  },
  "colorAmber700": Object {
    "color": "#b45309",
  },
  "colorAmber800": Object {
    "color": "#92400e",
  },
  "colorAmber900": Object {
    "color": "#78350f",
  },
  "colorBlack": Object {
    "color": "#000",
  },
  "colorBlue": Object {
    "color": "#3b82f6",
  },
  "colorBlue100": Object {
    "color": "#dbeafe",
  },
  "colorBlue200": Object {
    "color": "#bfdbfe",
  },
  "colorBlue300": Object {
    "color": "#93c5fd",
  },
  "colorBlue400": Object {
    "color": "#60a5fa",
  },
  "colorBlue50": Object {
    "color": "#eff6ff",
  },
  "colorBlue500": Object {
    "color": "#3b82f6",
  },
  "colorBlue600": Object {
    "color": "#2563eb",
  },
  "colorBlue700": Object {
    "color": "#1d4ed8",
  },
  "colorBlue800": Object {
    "color": "#1e40af",
  },
  "colorBlue900": Object {
    "color": "#1e3a8a",
  },
  "colorBlueGray": Object {
    "color": "#64748b",
  },
  "colorBlueGray100": Object {
    "color": "#f1f5f9",
  },
  "colorBlueGray200": Object {
    "color": "#e2e8f0",
  },
  "colorBlueGray300": Object {
    "color": "#cbd5e1",
  },
  "colorBlueGray400": Object {
    "color": "#94a3b8",
  },
  "colorBlueGray50": Object {
    "color": "#f8fafc",
  },
  "colorBlueGray500": Object {
    "color": "#64748b",
  },
  "colorBlueGray600": Object {
    "color": "#475569",
  },
  "colorBlueGray700": Object {
    "color": "#334155",
  },
  "colorBlueGray800": Object {
    "color": "#1e293b",
  },
  "colorBlueGray900": Object {
    "color": "#0f172a",
  },
  "colorCoolGray": Object {
    "color": "#6b7280",
  },
  "colorCoolGray100": Object {
    "color": "#f3f4f6",
  },
  "colorCoolGray200": Object {
    "color": "#e5e7eb",
  },
  "colorCoolGray300": Object {
    "color": "#d1d5db",
  },
  "colorCoolGray400": Object {
    "color": "#9ca3af",
  },
  "colorCoolGray50": Object {
    "color": "#f9fafb",
  },
  "colorCoolGray500": Object {
    "color": "#6b7280",
  },
  "colorCoolGray600": Object {
    "color": "#4b5563",
  },
  "colorCoolGray700": Object {
    "color": "#374151",
  },
  "colorCoolGray800": Object {
    "color": "#1f2937",
  },
  "colorCoolGray900": Object {
    "color": "#111827",
  },
  "colorCyan": Object {
    "color": "#06b6d4",
  },
  "colorCyan100": Object {
    "color": "#cffafe",
  },
  "colorCyan200": Object {
    "color": "#a5f3fc",
  },
  "colorCyan300": Object {
    "color": "#67e8f9",
  },
  "colorCyan400": Object {
    "color": "#22d3ee",
  },
  "colorCyan50": Object {
    "color": "#ecfeff",
  },
  "colorCyan500": Object {
    "color": "#06b6d4",
  },
  "colorCyan600": Object {
    "color": "#0891b2",
  },
  "colorCyan700": Object {
    "color": "#0e7490",
  },
  "colorCyan800": Object {
    "color": "#155e75",
  },
  "colorCyan900": Object {
    "color": "#164e63",
  },
  "colorEmerald": Object {
    "color": "#10b981",
  },
  "colorEmerald100": Object {
    "color": "#d1fae5",
  },
  "colorEmerald200": Object {
    "color": "#a7f3d0",
  },
  "colorEmerald300": Object {
    "color": "#6ee7b7",
  },
  "colorEmerald400": Object {
    "color": "#34d399",
  },
  "colorEmerald50": Object {
    "color": "#ecfdf5",
  },
  "colorEmerald500": Object {
    "color": "#10b981",
  },
  "colorEmerald600": Object {
    "color": "#059669",
  },
  "colorEmerald700": Object {
    "color": "#047857",
  },
  "colorEmerald800": Object {
    "color": "#065f46",
  },
  "colorEmerald900": Object {
    "color": "#064e3b",
  },
  "colorFuchsia": Object {
    "color": "#d946ef",
  },
  "colorFuchsia100": Object {
    "color": "#fae8ff",
  },
  "colorFuchsia200": Object {
    "color": "#f5d0fe",
  },
  "colorFuchsia300": Object {
    "color": "#f0abfc",
  },
  "colorFuchsia400": Object {
    "color": "#e879f9",
  },
  "colorFuchsia50": Object {
    "color": "#fdf4ff",
  },
  "colorFuchsia500": Object {
    "color": "#d946ef",
  },
  "colorFuchsia600": Object {
    "color": "#c026d3",
  },
  "colorFuchsia700": Object {
    "color": "#a21caf",
  },
  "colorFuchsia800": Object {
    "color": "#86198f",
  },
  "colorFuchsia900": Object {
    "color": "#701a75",
  },
  "colorGray": Object {
    "color": "#6b7280",
  },
  "colorGray100": Object {
    "color": "#f3f4f6",
  },
  "colorGray200": Object {
    "color": "#e5e7eb",
  },
  "colorGray300": Object {
    "color": "#d1d5db",
  },
  "colorGray400": Object {
    "color": "#9ca3af",
  },
  "colorGray50": Object {
    "color": "#f9fafb",
  },
  "colorGray500": Object {
    "color": "#6b7280",
  },
  "colorGray600": Object {
    "color": "#4b5563",
  },
  "colorGray700": Object {
    "color": "#374151",
  },
  "colorGray800": Object {
    "color": "#1f2937",
  },
  "colorGray900": Object {
    "color": "#111827",
  },
  "colorGreen": Object {
    "color": "#22c55e",
  },
  "colorGreen100": Object {
    "color": "#dcfce7",
  },
  "colorGreen200": Object {
    "color": "#bbf7d0",
  },
  "colorGreen300": Object {
    "color": "#86efac",
  },
  "colorGreen400": Object {
    "color": "#4ade80",
  },
  "colorGreen50": Object {
    "color": "#f0fdf4",
  },
  "colorGreen500": Object {
    "color": "#22c55e",
  },
  "colorGreen600": Object {
    "color": "#16a34a",
  },
  "colorGreen700": Object {
    "color": "#15803d",
  },
  "colorGreen800": Object {
    "color": "#166534",
  },
  "colorGreen900": Object {
    "color": "#14532d",
  },
  "colorIndigo": Object {
    "color": "#6366f1",
  },
  "colorIndigo100": Object {
    "color": "#e0e7ff",
  },
  "colorIndigo200": Object {
    "color": "#c7d2fe",
  },
  "colorIndigo300": Object {
    "color": "#a5b4fc",
  },
  "colorIndigo400": Object {
    "color": "#818cf8",
  },
  "colorIndigo50": Object {
    "color": "#eef2ff",
  },
  "colorIndigo500": Object {
    "color": "#6366f1",
  },
  "colorIndigo600": Object {
    "color": "#4f46e5",
  },
  "colorIndigo700": Object {
    "color": "#4338ca",
  },
  "colorIndigo800": Object {
    "color": "#3730a3",
  },
  "colorIndigo900": Object {
    "color": "#312e81",
  },
  "colorLightBlue": Object {
    "color": "#0ea5e9",
  },
  "colorLightBlue100": Object {
    "color": "#e0f2fe",
  },
  "colorLightBlue200": Object {
    "color": "#bae6fd",
  },
  "colorLightBlue300": Object {
    "color": "#7dd3fc",
  },
  "colorLightBlue400": Object {
    "color": "#38bdf8",
  },
  "colorLightBlue50": Object {
    "color": "#f0f9ff",
  },
  "colorLightBlue500": Object {
    "color": "#0ea5e9",
  },
  "colorLightBlue600": Object {
    "color": "#0284c7",
  },
  "colorLightBlue700": Object {
    "color": "#0369a1",
  },
  "colorLightBlue800": Object {
    "color": "#075985",
  },
  "colorLightBlue900": Object {
    "color": "#0c4a6e",
  },
  "colorLime": Object {
    "color": "#84cc16",
  },
  "colorLime100": Object {
    "color": "#ecfccb",
  },
  "colorLime200": Object {
    "color": "#d9f99d",
  },
  "colorLime300": Object {
    "color": "#bef264",
  },
  "colorLime400": Object {
    "color": "#a3e635",
  },
  "colorLime50": Object {
    "color": "#f7fee7",
  },
  "colorLime500": Object {
    "color": "#84cc16",
  },
  "colorLime600": Object {
    "color": "#65a30d",
  },
  "colorLime700": Object {
    "color": "#4d7c0f",
  },
  "colorLime800": Object {
    "color": "#3f6212",
  },
  "colorLime900": Object {
    "color": "#365314",
  },
  "colorOrange": Object {
    "color": "#f97316",
  },
  "colorOrange100": Object {
    "color": "#ffedd5",
  },
  "colorOrange200": Object {
    "color": "#fed7aa",
  },
  "colorOrange300": Object {
    "color": "#fdba74",
  },
  "colorOrange400": Object {
    "color": "#fb923c",
  },
  "colorOrange50": Object {
    "color": "#fff7ed",
  },
  "colorOrange500": Object {
    "color": "#f97316",
  },
  "colorOrange600": Object {
    "color": "#ea580c",
  },
  "colorOrange700": Object {
    "color": "#c2410c",
  },
  "colorOrange800": Object {
    "color": "#9a3412",
  },
  "colorOrange900": Object {
    "color": "#7c2d12",
  },
  "colorPink": Object {
    "color": "#ec4899",
  },
  "colorPink100": Object {
    "color": "#fce7f3",
  },
  "colorPink200": Object {
    "color": "#fbcfe8",
  },
  "colorPink300": Object {
    "color": "#f9a8d4",
  },
  "colorPink400": Object {
    "color": "#f472b6",
  },
  "colorPink50": Object {
    "color": "#fdf2f8",
  },
  "colorPink500": Object {
    "color": "#ec4899",
  },
  "colorPink600": Object {
    "color": "#db2777",
  },
  "colorPink700": Object {
    "color": "#be185d",
  },
  "colorPink800": Object {
    "color": "#9d174d",
  },
  "colorPink900": Object {
    "color": "#831843",
  },
  "colorPurple": Object {
    "color": "#8b5cf6",
  },
  "colorPurple100": Object {
    "color": "#ede9fe",
  },
  "colorPurple200": Object {
    "color": "#ddd6fe",
  },
  "colorPurple300": Object {
    "color": "#c4b5fd",
  },
  "colorPurple400": Object {
    "color": "#a78bfa",
  },
  "colorPurple50": Object {
    "color": "#f5f3ff",
  },
  "colorPurple500": Object {
    "color": "#8b5cf6",
  },
  "colorPurple600": Object {
    "color": "#7c3aed",
  },
  "colorPurple700": Object {
    "color": "#6d28d9",
  },
  "colorPurple800": Object {
    "color": "#5b21b6",
  },
  "colorPurple900": Object {
    "color": "#4c1d95",
  },
  "colorRed": Object {
    "color": "#ef4444",
  },
  "colorRed100": Object {
    "color": "#fee2e2",
  },
  "colorRed200": Object {
    "color": "#fecaca",
  },
  "colorRed300": Object {
    "color": "#fca5a5",
  },
  "colorRed400": Object {
    "color": "#f87171",
  },
  "colorRed50": Object {
    "color": "#fef2f2",
  },
  "colorRed500": Object {
    "color": "#ef4444",
  },
  "colorRed600": Object {
    "color": "#dc2626",
  },
  "colorRed700": Object {
    "color": "#b91c1c",
  },
  "colorRed800": Object {
    "color": "#991b1b",
  },
  "colorRed900": Object {
    "color": "#7f1d1d",
  },
  "colorRose": Object {
    "color": "#f43f5e",
  },
  "colorRose100": Object {
    "color": "#ffe4e6",
  },
  "colorRose200": Object {
    "color": "#fecdd3",
  },
  "colorRose300": Object {
    "color": "#fda4af",
  },
  "colorRose400": Object {
    "color": "#fb7185",
  },
  "colorRose50": Object {
    "color": "#fff1f2",
  },
  "colorRose500": Object {
    "color": "#f43f5e",
  },
  "colorRose600": Object {
    "color": "#e11d48",
  },
  "colorRose700": Object {
    "color": "#be123c",
  },
  "colorRose800": Object {
    "color": "#9f1239",
  },
  "colorRose900": Object {
    "color": "#881337",
  },
  "colorTeal": Object {
    "color": "#14b8a6",
  },
  "colorTeal100": Object {
    "color": "#ccfbf1",
  },
  "colorTeal200": Object {
    "color": "#99f6e4",
  },
  "colorTeal300": Object {
    "color": "#5eead4",
  },
  "colorTeal400": Object {
    "color": "#2dd4bf",
  },
  "colorTeal50": Object {
    "color": "#f0fdfa",
  },
  "colorTeal500": Object {
    "color": "#14b8a6",
  },
  "colorTeal600": Object {
    "color": "#0d9488",
  },
  "colorTeal700": Object {
    "color": "#0f766e",
  },
  "colorTeal800": Object {
    "color": "#115e59",
  },
  "colorTeal900": Object {
    "color": "#134e4a",
  },
  "colorTransparent": Object {
    "color": "transparent",
  },
  "colorTrueGray": Object {
    "color": "#737373",
  },
  "colorTrueGray100": Object {
    "color": "#f5f5f5",
  },
  "colorTrueGray200": Object {
    "color": "#e5e5e5",
  },
  "colorTrueGray300": Object {
    "color": "#d4d4d4",
  },
  "colorTrueGray400": Object {
    "color": "#a3a3a3",
  },
  "colorTrueGray50": Object {
    "color": "#fafafa",
  },
  "colorTrueGray500": Object {
    "color": "#737373",
  },
  "colorTrueGray600": Object {
    "color": "#525252",
  },
  "colorTrueGray700": Object {
    "color": "#404040",
  },
  "colorTrueGray800": Object {
    "color": "#262626",
  },
  "colorTrueGray900": Object {
    "color": "#171717",
  },
  "colorViolet": Object {
    "color": "#8b5cf6",
  },
  "colorViolet100": Object {
    "color": "#ede9fe",
  },
  "colorViolet200": Object {
    "color": "#ddd6fe",
  },
  "colorViolet300": Object {
    "color": "#c4b5fd",
  },
  "colorViolet400": Object {
    "color": "#a78bfa",
  },
  "colorViolet50": Object {
    "color": "#f5f3ff",
  },
  "colorViolet500": Object {
    "color": "#8b5cf6",
  },
  "colorViolet600": Object {
    "color": "#7c3aed",
  },
  "colorViolet700": Object {
    "color": "#6d28d9",
  },
  "colorViolet800": Object {
    "color": "#5b21b6",
  },
  "colorViolet900": Object {
    "color": "#4c1d95",
  },
  "colorWarmGray": Object {
    "color": "#78716c",
  },
  "colorWarmGray100": Object {
    "color": "#f5f5f4",
  },
  "colorWarmGray200": Object {
    "color": "#e7e5e4",
  },
  "colorWarmGray300": Object {
    "color": "#d6d3d1",
  },
  "colorWarmGray400": Object {
    "color": "#a8a29e",
  },
  "colorWarmGray50": Object {
    "color": "#fafaf9",
  },
  "colorWarmGray500": Object {
    "color": "#78716c",
  },
  "colorWarmGray600": Object {
    "color": "#57534e",
  },
  "colorWarmGray700": Object {
    "color": "#44403c",
  },
  "colorWarmGray800": Object {
    "color": "#292524",
  },
  "colorWarmGray900": Object {
    "color": "#1c1917",
  },
  "colorWhite": Object {
    "color": "#fff",
  },
  "colorYellow": Object {
    "color": "#f59e0b",
  },
  "colorYellow100": Object {
    "color": "#fef3c7",
  },
  "colorYellow200": Object {
    "color": "#fde68a",
  },
  "colorYellow300": Object {
    "color": "#fcd34d",
  },
  "colorYellow400": Object {
    "color": "#fbbf24",
  },
  "colorYellow50": Object {
    "color": "#fffbeb",
  },
  "colorYellow500": Object {
    "color": "#f59e0b",
  },
  "colorYellow600": Object {
    "color": "#d97706",
  },
  "colorYellow700": Object {
    "color": "#b45309",
  },
  "colorYellow800": Object {
    "color": "#92400e",
  },
  "colorYellow900": Object {
    "color": "#78350f",
  },
  "lineLineThrough": Object {
    "textDecorationLine": "line-through",
  },
  "lineNone": Object {
    "textDecorationLine": "none",
  },
  "lineUnderline": Object {
    "textDecorationLine": "underline",
  },
  "lineUnderlineLineThrough": Object {
    "textDecorationLine": "underline line-through",
  },
  "text2Xl": Object {
    "fontSize": 24,
    "lineHeight": 32,
  },
  "text3Xl": Object {
    "fontSize": 30,
    "lineHeight": 36,
  },
  "text4Xl": Object {
    "fontSize": 36,
    "lineHeight": 40,
  },
  "textBase": Object {
    "fontSize": 16,
    "lineHeight": 24,
  },
  "textLg": Object {
    "fontSize": 18,
    "lineHeight": 28,
  },
  "textSm": Object {
    "fontSize": 14,
    "lineHeight": 20,
  },
  "textXl": Object {
    "fontSize": 20,
    "lineHeight": 28,
  },
  "textXs": Object {
    "fontSize": 12,
    "lineHeight": 16,
  },
  "transformCapitalize": Object {
    "textTransform": "capitalize",
  },
  "transformLowercase": Object {
    "textTransform": "lowercase",
  },
  "transformNone": Object {
    "textTransform": "none",
  },
  "transformUppercase": Object {
    "textTransform": "uppercase",
  },
}
`;
